src/tools/reloading.tsx

```diff
@@ -89,31 +88,44 @@ const demoData = (match, counter, perPage) => {
   }
 }
 
-const fallbackData = (match) => {
+const fallbackData = (match: any) => {
   switch (match.path) {
     case '/fundraising-events/:id/last-donation':
     case '/fundraising-events/:id/top-donation':
     case '/fundraising-events/:id/top-donor':
     case '/fundraising-events/:id/last-comment':
-      return { author: { name: "-" } }
+      return { author: { name: '-' } }
     default:
       return null
   }
 }
 
-export function reloading(WrappedComponent) {
-  return class extends React.Component {
-    constructor(props) {
-      super(props);
+export function reloading(WrappedComponent: any) {
+  type State = {
+    params: URLSearchParams
+    data: any
+    listData: any[]
+    counter: number
+    demo: string | null
+    maxCount: number
+    interval: number
+    listMode: boolean
+    since: string | null
+    hashtags: string | null
+  }
+  return class ReloadingClass extends React.Component<RouteComponentProps, State> {
+    interval: any
+    constructor(props: any) {
+      super(props)
       const params = new URLSearchParams(this.props.location.search)
       this.state = {
         params,
         data: null,
         listData: [],
         counter: 1,
         demo: params.get('demo'),
-        maxCount: parseInt(params.get('maxCount'), 10) || 1,
-        interval: parseInt(params.get('interval'), 10) || 3,
+        maxCount: params.get('maxCount') ? parseInt(params.get('maxCount') as string, 10) : 1,
+        interval: params.get('interval') ? parseInt(params.get('interval') as string, 10) : 3,
```

Comment:
```suggestion
        maxCount: parseInt(params.get('maxCount') || '', 10) || 1,
        interval: parseInt(params.get('interval') || '', 10) || 3,
```

Reply:
I just had a similar thought.
I don't think casting to a string makes sense here, because it actually can be null. My idea was to put the fallback case inside of parseInt instead? (It's kinda weird to add it as a string just to parse it back to an integer again, though)

```suggestion
        maxCount: parseInt(params.get('maxCount') || '1', 10),
        interval: parseInt(params.get('interval') || '3', 10),
```